### ISSUE TITLE
DOC: reading nybb dataset is failing with encoding error on readthedocs

### ISSUE THREAD

**Issue by jpeacock29** [geopandas/geopandas#629]
The example on the [Geocoding](http://geopandas.readthedocs.io/en/latest/geocoding.html) page appears to have failed when run and does not give the correct result.

**Comment by jorisvandenbossche** [comment]
Thanks for the report! 
It's not an error with the geocoding itself, but an installation issue with fiona.

**Comment by jorisvandenbossche** [comment]
So the error is:

```
In [1]: boros = gpd.read_file(gpd.datasets.get_path("nybb"))
...
fiona/_err.pyx in fiona._err.GDALErrCtxManager.__exit__()
CPLE_AppDefinedError: b'Recode from CP437 to UTF-8 failed with the error: "Invalid argument".'
```

I cannot reproduce this locally (although I already have seen it locally as well at some point, but can't remember any more ..)

The environment.yml file that readthedocs uses to create the environment is here: https://github.com/geopandas/geopandas/blob/master/doc/environment.yml

Potentially related issues (with similar error): https://github.com/Toblerity/Fiona/issues/455, https://github.com/conda-forge/gdal-feedstock/issues/83

@ocefpaf any idea where to look?

**Comment by jorisvandenbossche** [comment]
> (although I already have seen it locally as well at some point, but can't remember any more ..)

OK, I now remember again: I get this in my root environment, but this is a mishmash of a lot of packages added over the years, so difficult to pinpoint (although all relevant packages come from conda-forge). Can't reproduce in a clean env.

**Comment by sgillies** [comment]
@jorisvandenbossche I peeked in the nybb.dbf and it seems that its encoding is plain old ascii. I don't know why GDAL tries to recode it based on cp437 (which happens to be the next number in the list of code pages). I think that there should be a closer look at whether the GDAL from conda-forge and iconv are properly configured.

**Comment by jorisvandenbossche** [comment]
So in the fresh environment I have (installed just now):

```
gdal                      2.1.3                    py35_7    conda-forge
# no libgdal strangely
libiconv                  1.15                          0    conda-forge
```

and in my root (where it is failing):

```
gdal                      2.1.4                    py35_1    conda-forge
libgdal                   2.1.4                         2    conda-forge
libiconv                  1.14                          4    conda-forge
```

On readthedocs it seemed to be downloading libgdal 2.1.4 and libiconv 1.15 (3 weeks ago, https://readthedocs.org/projects/geopandas/builds/6313982/), I triggered it now again and it installed gdal 2.1.3 and libiconv 1.15 (as in my local environment that works), but now it is failing on geos .. (https://readthedocs.org/projects/geopandas/builds/6419946/)

@ocefpaf any idea why there was a gdal 2.1.4 and now you get 2.1.3 ? 
It is possible that readthedocs 3 weeks ago just had bad luck that it downloaded at a moment the different builds were under way and not fully synchronized yet throughout conda-forge.

**Comment by jdmcbr** [comment]
Closed by #641.

**Comment by jorisvandenbossche** [comment]
The broader issue (the encoding error, I also had it locally) is not closed by #641, but it is also not an issue of geopandas itself, so fine to close.

**Comment by jdmcbr** [comment]
The encoding issue has been addressed in fiona as of 1.7.11 (you may need to specify encoding explicitly if there's a missing or incorrectly specified cpg file); I think its appearance here was related to the issues setting up the environment on readthedocs.
